xdsl/dialects/builtin.py

```diff
@@ -1175,8 +1175,18 @@ def from_strings(name: str, value: str, type: Attribute = NoneAttr()) -> OpaqueA
         return OpaqueAttr([StringAttr(name), StringAttr(value), type])
 
 
+class MemrefLayoutAttr(Attribute):
```

Comment:
```suggestion
class MemrefLayoutAttr(Attribute, abc.ABC):
```

Reply:
Just doing it, but why? It is not declaring any abstract method, what's the point?

Reply:
Seems like a more protective default, if it's not designed to be initialized, then why not make that throw an error at runtime?

Reply:
Oh, forgot about the instanciation guard. Good reminder, thanks!